pkg/cli/helm/installer.go

```diff
@@ -0,0 +1,351 @@
+	URL string
+	// Version of the helm chart to install.
+	// If loading from a directory, needs to match the chart version.
+	Version string
+	// Name of the Helm chart to install.
+	// Required only if pulling from the specified URL.
+	Name string
+}
+
+// Init initializes the Installer with the specified options.
+func (i *Installer) Init(kubeconfigPath string, o ChartOptions) error {
+	if o.Directory == "" && o.URL == "" {
+		return errors.New("either chart directory or URL must be set")
+	}
+
+	if o.Version == "" {
+		return errors.New("chart version must be set")
+	}
+
+	chart, err := resolveHelmChart(o.Version, o.Name, o.URL, o.Directory)
+	if err != nil {
+		return fmt.Errorf("failed to resolve Helm chart: %w", err)
+	}
+	i.chart = chart
+
+	cfg, err := newActionsCfg(i.ReleaseNamespace, kubeconfigPath)
+	if err != nil {
+		return fmt.Errorf("failed to create Helm action configuration: %w", err)
+	}
+	i.cfg = cfg
+	return nil
+}
+
+// Install the Helm chart.
+// Calling Install multiple times is idempotent; it will re-apply the manifests using upgrade.
+func (i *Installer) Install(ctx context.Context) error {
+	rel, err := action.NewGet(i.cfg).Run(i.ReleaseName)
+	if err != nil {
+		// Release does not exist, install it.
+		if errors.Is(err, driver.ErrReleaseNotFound) {
+			return i.install(ctx)
+		}
+	}
+	// If the release already exists, we will re-apply the manifests using upgrade.
+	// We're not actually upgrading to a new version, but using upgrade to re-apply manifests.
+	// This is how Helm expects us to re-apply manifests.
+	// To prevent accidental version upgrades, we will explicitly check that the resolved chart version matches the installed chart version.
+	if i.chart.Metadata.Version != rel.Chart.Metadata.Version {
+		return fmt.Errorf("cannot overwrite existing release with a different chart version. Expected %s, got %s",
+			rel.Chart.Metadata.Version, i.chart.Metadata.Version,
+		)
+	}
+	return i.Upgrade(ctx)
+}
+
+// GetRelease gets the installed Helm release.
+func (i *Installer) GetRelease() (*release.Release, error) {
+	if i.release == nil {
+		return nil, errors.New("chart not installed")
+	}
+	return i.release, nil
+}
+
+// RenderTemplates renders the Helm chart templates.
+func (i Installer) RenderTemplates(ctx context.Context, uninstallOrd bool) (helmutils.RenderedTemplates, error) {
+	i.DryRun = true
+	if err := i.Install(ctx); err != nil {
+		return nil, err
+	}
+	rendered := helmutils.RenderedTemplates{}
+	if err := rendered.FromString(i.release.Manifest, uninstallOrd); err != nil {
+		return nil, err
+	}
+	return rendered, nil
+}
+
+func (i *Installer) install(ctx context.Context) error {
+	install := action.NewInstall(i.cfg)
+	install.ReleaseName = i.ReleaseName
+	install.Namespace = i.ReleaseNamespace
+	install.DryRun = i.DryRun
+	install.CreateNamespace = i.CreateReleaseNamespace
+	install.Wait = false
+	install.TakeOwnership = true
+	install.DisableHooks = true
```

Comment:
:thinking: does this mean that we need to manage the IP approval process ourselves? Is this the reason for calling `installDBOperators` inside `provisionDBNamespace`? Why did you go with this approach?

Reply:
I answered here - https://github.com/percona/everest/pull/825#discussion_r1849470128